### PR TITLE
:memo: Declarative capsule spec deploy docs

### PR DESCRIPTION
**Pull request by Munklinde96**

### CODE REVIEW

File: docs/docs/getting-started/first-capsule.mdx [hunk 1]
```diff
@@ -109,3 +109,51 @@ rig deploy nginx-capsule -i nginx -r 3 --set-network-interface network.yaml --se
...
+
+```yaml title="nginx-capsule.yaml"
+apiVersion: platform.rig.dev/v1
+kind: Capsule
+environment: prod
+project: test
+spec:
+  files:
+  - path: /usr/share/nginx/html/index.html
+    string: <html>
```
Comment (andersjohnsen) [inline docs/docs/getting-started/first-capsule.mdx:129]: Does this work? I would expect a | here, ala:



```
    string: |
     <html>
       <body>
         <h1>Hello World!</h1>
       </body>
     </html>
```

File: docs/docs/getting-started/first-capsule.mdx [hunk 2]
```diff
@@ -109,3 +109,51 @@ rig deploy nginx-capsule -i nginx -r 3 --set-network-interface network.yaml --se
...
+              <body>
+                <h1>Hello World!</h1>
+              </body>
+            </html>
+  image: nginx:latest
+  interfaces:
+  - name: http
+    port: 80
+  scale:
```
Comment (andersjohnsen) [inline docs/docs/getting-started/first-capsule.mdx:138]: Add resource requests

File: docs/docs/getting-started/first-capsule.mdx [hunk 3]
```diff
@@ -109,3 +109,51 @@ rig deploy nginx-capsule -i nginx -r 3 --set-network-interface network.yaml --se
...
+spec:
+  files:
+  - path: /usr/share/nginx/html/index.html
+    string: <html>
+              <body>
+                <h1>Hello World!</h1>
+              </body>
+            </html>
+  image: nginx:latest
+  interfaces:
```
Comment (andersjohnsen) [inline docs/docs/getting-started/first-capsule.mdx:135]: Add readiness + liveness check

File: docs/docs/getting-started/first-capsule.mdx [hunk 4]
```diff
@@ -109,3 +109,51 @@ rig deploy nginx-capsule -i nginx -r 3 --set-network-interface network.yaml --se
...
+possible to deploy a capsule using a yaml file describing it.
+
+For example, we can create a `nginx-capsule.yaml` file that describes the capsule in the example above.
+Notice the project and environment fields, which are used to identify and place the capsule in Rig.
+
+```yaml title="nginx-capsule.yaml"
+apiVersion: platform.rig.dev/v1
+kind: Capsule
+environment: prod
+project: test
```
Comment (andersjohnsen) [inline docs/docs/getting-started/first-capsule.mdx:125]: test -> acme

File: docs/docs/getting-started/first-capsule.mdx [hunk 5]
```diff
@@ -109,3 +109,51 @@ rig deploy nginx-capsule -i nginx -r 3 --set-network-interface network.yaml --se
...
+Since a Capsule can be described declaratively, it is, similarly to `kubectl apply`,
+possible to deploy a capsule using a yaml file describing it.
+
+For example, we can create a `nginx-capsule.yaml` file that describes the capsule in the example above.
+Notice the project and environment fields, which are used to identify and place the capsule in Rig.
+
+```yaml title="nginx-capsule.yaml"
+apiVersion: platform.rig.dev/v1
+kind: Capsule
+environment: prod
```
Comment (andersjohnsen) [inline docs/docs/getting-started/first-capsule.mdx:124]: It's missing a name, ala `name: my-capsule`

**Review by andersjohnsen — APPROVED** [review]

File: docs/docs/getting-started/first-capsule.mdx [hunk 6]
```diff
@@ -109,3 +109,51 @@ rig deploy nginx-capsule -i nginx -r 3 --set-network-interface network.yaml --se
...
+  scale:
+    horizontal:
+      instances:
+        min: 2
+```
+
+It can then be deployed using the following command:
+
+```bash
+rig deploy nginx-capsule -f nginx-capsule.yaml
```
Comment (andersjohnsen) [inline docs/docs/getting-started/first-capsule.mdx:147]: name is not needed here anymore

**Review by andersjohnsen — APPROVED** [review]

File: docs/docs/getting-started/first-capsule.mdx (hunk 5 above)
Comment (Munklinde96) [reply docs/docs/getting-started/first-capsule.mdx:124]: Ye, but you give a name in the deploy command which takes precedence

File: docs/docs/getting-started/first-capsule.mdx (hunk 6 above)
Comment (Munklinde96) [reply docs/docs/getting-started/first-capsule.mdx:147]: i see

**Pull request merged**